src/main/java/org/mozilla/jss/util/NativeProxy.java

```diff
@@ -132,6 +134,7 @@ public int hashCode() {
      * @deprecated finalize() in Object has been deprecated. Use close(...)
      * from the AutoCloseable interface instead.
      */
+    @Override
```

Comment:
My understanding was that finalize wasn't strictly an override in all versions of JDK? I thought in the future (e.g., JDK17?) they were removing the finalize member on Object. Maybe I'm misremembering and they've not made plans to fully remove it? Might be worth looking into.

(Sorry, comment was on https://github.com/dogtagpki/jss/pull/724/files#r636855103 because that's the first PR I noticed this on).

Reply:
To my understanding there are issues with `finalize()` in general so it's no longer recommended to use and we should replace it before it's actually removed in the future. Right now we're still relying on the method as an extension of the `Object.finalize()`, so the `@Override` is simply affirming that dependency (and making sure that it's named correctly).

Reply:
I've not seen a hard plan to get rid of `Object.finalize()` but as you say it's bad form to depend on it, we should definitely consider not doing this.

Reply:
I believe it will be eventually removed in favor of https://docs.oracle.com/javase/9/docs/api/java/lang/ref/Cleaner.html